src/Database/LSMTree.hs

```diff
@@ -0,0 +1,551 @@
+-- TODO: document this module
```

Comment:
What's your plan for this?

For the individual function Haddocks, they can go out of sync, but we already have to consider keeping `Normal`/`Monoidal` in sync, so maybe having a third copy is not too bad?

And I would definitely add a comment to the module explaining that it's experimental and mainly for testing, as well as referring to the `Normal`/`Monoidal` version for full documentation. Also, a TODO to decide which public API(s) to provide in the end.

Reply:
We discussed this 1-on-1 already, but for now I think we should omit the documentation until we'

* decide how to deduplicate documentation between modules, or
* decide that we want to duplicate haddocks across different modules

I'll be more verbose in the TODO here

Reply:
Makes sense not to duplicate, but yes, in that case a short header explaining why this module exists (and doesn't have docs) is still good to have.